### PR TITLE
Moved digests methods and renamed them.

### LINKED ISSUE #5: Move digest functions
Move each of the digest functions from https://github.com/2start/bakker/blob/master/bakker/utils.py to the corresponding node in https://github.com/2start/bakker/blob/master/bakker/checkpoint.py respectively.

Rename to checksum at the same time.

### PR DESCRIPTION
**Pull request by eichenroth**
fixes #5

### CODE REVIEW

**Review by klaudworks — COMMENTED** [review]
Okay I think this is better than the previous solution. However, if we use a static method of the TreeNodes to create the TreeNode itself we should probably just create a static factory method inside each TreeNode that builds it.

**Comment by klaudworks** [comment]
Okay factory method might be the wrong description. I am talking about a static `build()` method. That calls the checksum method internally and calls the constructor with the checksum.

**Comment by eichenroth** [comment]
Agreed

File: bakker/checkpoint.py [hunk 1]
```diff
@@ -66,10 +55,39 @@ def to_dict(self):
...
+            message.update(child_digest)
+        checksum = message.hexdigest()
+
+        return DirectoryNode(name, checksum, file_permissions, children)
+
     @staticmethod
     def from_dict(d):
         return DirectoryNode(d['name'], d['checksum'], d['permissions'], {child['name']: TreeNode.from_dict(child) for child in d['children']})
 
+    @staticmethod
```
Comment (klaudworks) [inline bakker/checkpoint.py]: Concentration please.

**Review by klaudworks — CHANGES_REQUESTED** [review]

**Pull request merged**